src/internet_identity/src/state/temp_keys.rs

```diff
@@ -130,4 +122,6 @@ pub struct TempKeyExpiration {
     pub device_key: DeviceKey,
     /// The expiration timestamp of the temp key
     pub expiration: Timestamp,
+    /// The anchor the temporary key is linked to
+    anchor: AnchorNumber,
```

Comment:
nit: I'd put the anchor with the `device_key` or rename `device_key` to reflect that it's also the key used in the map